3_javascript/javascript1/week4/Homework/script.js

```diff
@@ -0,0 +1,188 @@
+window.speechSynthesis.onvoiceschanged = () => {
+  voices = window.speechSynthesis.getVoices();
+};
+
+// Speech to Text
+const SpeechRecognition =
+  window.SpeechRecognition || window.webkitSpeechRecognition;
+let toggleBtn = null;
+if (typeof SpeechRecognition === "undefined") {
+  startBtn.remove();
+  result.innerHTML =
+    "<strong>Your browser does not support Speech API. Please download latest Chrome version.<strong>";
+} else {
+  const recognition = new SpeechRecognition();
+  recognition.continuous = true; // If true, the silence period is longer
+  recognition.interimResults = true;
+  recognition.onresult = (event) => {
+    const current = event.resultIndex;
+    const recognitionResult = event.results[current];
+    const recognitionText = recognitionResult[0].transcript;
+
+    if (recognitionResult.isFinal) {
+      processing.innerHTML = "processing ...";
+
+      const reply = getReply(recognitionText);
+      const p = document.createElement("p");
+      p.innerHTML = `<strong>You said:</strong> ${recognitionText} </br><strong>Benjamin said:</strong> ${reply}`;
+      processing.innerHTML = "";
+      result.appendChild(p);
+
+      readOutLoud(reply);
+    } else {
+      processing.innerHTML = `listening: ${recognitionText}`;
+    }
+  };
+  let listening = false;
+  toggleBtn = () => {
+    if (listening) {
+      recognition.stop();
+      startBtn.textContent = "Start listening";
+    } else {
+      recognition.start();
+      startBtn.textContent = "Stop listening";
+    }
+    listening = !listening;
+  };
+  startBtn.addEventListener("click", toggleBtn);
+}
+
+// Processor
+function getReply(rawText) {
+  let text = rawText.replace(/\s/g, "").replace(/\'/g, "");
+  text = text.toLowerCase();
+  let reply = null;
+
+  if (text.includes("hello") || text.trim() == "hi" || text.includes("hey")) {
+    reply = getRandomItemFromArray(hello);
+  } else if (text.includes("Hello, my name is Angel")) {
+    reply = "Nice to meet you,Angel.";
+  } else if (text.includes("what is my name") || text.includes("human")) {
+    reply = "Angel";
+  } else if (
+    text.includes("Add fishing to my todo") ||
+    text.includes("baking") ||
+    text.includes("Painting")
+  ) {
+    reply = "fishing added to your todo";
+  } else if (
+    text.includes("Add singing in the shower to my todo") ||
+    text.includes("baking") ||
+    text.includes("Painting") ||
+    text.includes("fishing")
+  ) {
+    reply = "singing added to your todo";
+  } else if (
+    text.includes("remove fishing from my todo") ||
+    text.includes("baking") ||
+    text.includes("Painting") ||
+    text.includes("singing")
+  ) {
+    reply = "fishing removed from my todo";
+  } else if (
+    text.includes("What is on my todo?") ||
+    text.includes("baking") ||
+    text.includes("Painting") ||
+    text.includes("singing")
+  ) {
+    reply = "You have 3 todos";
+  } else if (text.includes("whattimeisit")) {
```

Comment:
here you are checking the text without blank spaces but in others cases you are checking also texts with spaces, both wont work